coordinape/coordinape
# Add circle ID to URLs

## Motivation and Context

Put the circle ID in the URL and use that as the single source of truth for what circle should be displayed on a given page. This removes the need to "guess" the current circle, which makes it easier to bridge old (fetchManifest-based) and new (Zeus-based) pages, and addresses some bugs (#951, #981, et al.).

## Description

- add circle ID to URL for all circle-specific routes
- don't store circle ID in local storage
- rework auth flow so it's completed before any React Router Routes are rendered

## Test and Deployment Plan

- load all circle-specific pages, both from within the app and as the first page in the browser tab
- try loading circle-specific pages and circle-admin-only pages from scratch while logged in and logged out
- try out all steps in login flow (login, logout, cancel signature prompt, use Wallet Connect, etc)
- verify that Dework integration goes to the right callback page
